task/mgmt-gateway/src/mgs_gimlet.rs

```diff
@@ -80,15 +95,18 @@ impl MgsHandler {
         // avoid blocking while the entire erase happens. If we're in that case,
         // set our timer for 1 tick from now to give a window for other
         // interrupts/notifications to arrive.
-        if self.host_flash_update.needs_sectors_erased() {
+        if self.host_flash_update.is_preparing()
+            || self.sp_update.is_preparing()
+        {
             Some(sys_get_timer().now + 1)
```

Comment:
That's intentional - if we're "preparing" (e.g., erasing flash), MGS is actively waiting for us to finish so it can start sending us an update. We only take a pause here to check for any new interrupts (e.g., MGS requesting status / progress). If we could remain responsive without this timer it wouldn't be here at all and we'd be doing all the prep work in a loop.

Reply:
Makes sense!